### ISSUE TITLE
Add full support to render PGP/MIME

### ISSUE THREAD

**Issue by sosnovsky** [FlowCrypt/flowcrypt-android#2527]
Android:
<img width="320" src="https://github.com/FlowCrypt/flowcrypt-android/assets/1062093/8a2ffc3c-4c03-4870-9ac5-a4fc6ce5ae90">

Browser extension:
<img width="600" alt="Screenshot 2023-12-12 at 22 42 59" src="https://github.com/FlowCrypt/flowcrypt-android/assets/1062093/05959250-736b-4d97-b2de-13c852ec93f3">

**Comment by DenBond7** [comment]
https://datatracker.ietf.org/doc/html/rfc3156#section-4

**Comment by DenBond7** [comment]
@sosnovsky Do you know what software was used to create this message? It looks like it doesn't follow the PGP/mime standard.

**Comment by sosnovsky** [comment]
Hmm, I think it was created with browser extension using PGP/MIME mode in Firefox

**Comment by sosnovsky** [comment]
Yeah, looks like we should update code for PGP/MIME message composing in browser extension - I tried to send similar message from protonmail and it's rendered in Android app. But maybe it will be a simple fix for Android, as I opened the same message on iOS and it was rendered there (maybe `no name` for image causes problem on Android):

<img src="https://github.com/FlowCrypt/flowcrypt-android/assets/1062093/8895b646-9107-4a27-8041-f44917527275" width="300">

I've just noticed another issue in message sent from proton (`PGP/MIME inline image for flowcrypt-android #2527` in `flowcrypt.compatibility` account) - message has inline image (138kb) attached, but in attachments list there is `.sig` file (310b), and there are also 2 empty blocks with gray border:

<img src="https://github.com/FlowCrypt/flowcrypt-android/assets/1062093/c8e49858-dc0e-4078-a49a-7b6679706b12" width="300">

**Comment by DenBond7** [comment]
> Yeah, looks like we should update code for PGP/MIME message composing in browser extension 
> But maybe it will be a simple fix for Android, as I opened the same message on iOS and it was rendered there (maybe no name for image causes problem on Android):

Yes, I can add a fix. But this will be a non-standard way, and it will significantly complicate the code.

The current browser realization doesn't follow the standard. Let me add more details here

The browser extension generates the following:

```
------sinikael-?=_1-16998840995020.4112130840361884
Content-Type: application/pgp-encrypted; name=
Content-Description: PGP/MIME version identification
Content-Disposition: attachment
X-Attachment-Id: f_mxDfixWgPOEcEygnCeJiNRdWbtvpqe@flowcrypt
Content-Id: <f_mxDfixWgPOEcEygnCeJiNRdWbtvpqe@flowcrypt>
Content-Transfer-Encoding: 7bit

Version: 1
```

Based on https://proton.me/support/pgp-mime-pgp-inline it should be something like this one:

![image](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/513957d3-fb97-442f-9ec3-e53b528f13a1)

Also, based on https://datatracker.ietf.org/doc/html/rfc3156#autoid-4

```
The multipart/encrypted MIME body MUST consist of exactly two body
   parts, the first with content type "application/pgp-encrypted".  This
   body contains the control information.  A message complying with this
   standard MUST contain a "Version: 1" field in this body.  Since the
   OpenPGP packet format contains all other information necessary for
   decrypting, NO OTHER INFORMATION IS REQUIRED HERE.
```

![image](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/c978678d-e295-4c37-912c-bd6bfda75695)

All of that means we have to generate the following:

```
------sinikael-?=_1-16998840995020.4112130840361884
Content-Type: application/pgp-encrypted; 
Content-Description: PGP/MIME version identification
Content-Transfer-Encoding: 7bit

Version: 1
```
###########################################################

For now, on my side **name=** is a problem. JavaMail recognizes it as an error. If a parameter is present it should have a value(or `quoted-string` for the empty value like `name=""`). More details can be found [here](https://datatracker.ietf.org/doc/html/rfc2045#section-5.1)

![image](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/83ff6301-430a-4f69-b472-f5280dd2afe6)

###########################################################
> I tried to send similar message from protonmail and it's rendered in Android app.

It is not `PGP/MIME` encrypted message. It's `PGP/Inline`

![image](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/e929cee0-5ac3-46b1-b089-c1b239514a24)

Like 

![image](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/2448a174-92ee-4082-944d-029fd7164e90)

**Comment by sosnovsky** [comment]
> Yes, I can add a fix. But this will be a non-standard way, and it will significantly complicate the code.

Got it, thanks for detailed explanation, then we'll fix it on browser extension side, so PGP/MIME messages will be formatted according to standards. I think we had the same issue on FES and updated messages format there too, probably will be helpful in fixing issue in the extension code

**Comment by DenBond7** [comment]
> I think we had the same issue on FES and updated messages format there too, probably will be helpful in fixing issue in the extension code

Yup. You are right. https://github.com/FlowCrypt/enterprise-server/issues/4204

**Comment by sosnovsky** [comment]
Hi @DenBond7, I tried to send PGP/MIME message from browser extension with updated structure from mentioned post, but it still fails to decrypt on Android (`PGP/MIME message for flowcrypt-android #2527` message in `flowcrypt.compatibility` account:

<img src="https://github.com/FlowCrypt/flowcrypt-android/assets/1062093/0a4a330c-89ac-438b-b43f-ac7b691d661e" width="300">

Can you please re-check what causes error there? Should I make some changes in extension code or it needs to be fixed on Android side?

**Comment by DenBond7** [comment]
> Should I make some changes in extension code or it needs to be fixed on Android side?

I think we can discuss it. I've checked -  you generate a MIME message almost correctly. But maybe some things can be dropped. Let me add more details.(I've changed some things for simplicity)

![image](https://github.com/FlowCrypt/flowcrypt-android/assets/2863246/6f3193ed-99fe-4870-a005-158b1d0f453e)

I think it should looks like this one 
```
------sinikael-?=_1-17036833722940.42154288944199925
Content-Type: application/pgp-encrypted
Content-Description: PGP/MIME version identification
Content-Transfer-Encoding: 7bit

Version: 1

------sinikael-?=_1-17036833722940.42154288944199925
Content-Type: application/octet-stream; name=encrypted.asc
Content-Disposition: inline; filename=encrypted.asc
Content-Description: OpenPGP encrypted message
Content-Transfer-Encoding: 7bit
```

1. Need to add a new line after `Version: 1` text. [RFC](https://datatracker.ietf.org/doc/html/rfc3156#autoid-4) and [here](https://datatracker.ietf.org/doc/html/rfc2046#section-5.1.1) 
2. IMHO, based on [RFC](https://datatracker.ietf.org/doc/html/rfc2045#section-7) `Content-Id` can be dropped here as redundant.
3. `X-Attachment-Id` can be dropped too.

I think 1 is important. But 2 and 3 can be discussed. As for me, if it is possible I would drop them.

If you remove `Content-Id` and `X-Attachment-Id` a message will be rendered. But the Android side should be fixed anyway because it is not clear enough.
